src/client/app/effects/snippet.ts

```diff
@@ -187,6 +188,48 @@ export class SnippetEffects {
         .map((updatedSnippet) => new Snippet.SaveAction(updatedSnippet))
         .catch(exception => Observable.of(new UI.ReportErrorAction(Strings().snippetUpdateError, exception)));
 
+    @Effect({ dispatch: false })
+    openInPlayground$: Observable<Action> = this.actions$
+        .ofType(Snippet.SnippetActionTypes.OPEN_IN_PLAYGROUND)
+        .map(action => action.payload)
+        .map(payload => {
+            if (Utilities.platform === PlatformType.IOS) {
+                AI.trackEvent('Unsupported open in playground');
+                return;
+            }
+            let { type, id, isDownload } = payload;
+            let handler, extension;
+            switch (environment.current.host.toUpperCase()) {
+                case HostType.EXCEL:
+                    handler = 'ms-excel:ofe|u|';
+                    extension = '.xlsx';
+                    break;
+                case HostType.WORD:
+                    handler = 'ms-word:ofe|u|';
+                    extension = '.docx';
+                    break;
+                case HostType.POWERPOINT:
+                    case HostType.WORD:
+                    handler = 'ms-powerpoint:ofe|u|';
+                    extension = '.pptx';
+                    break;
+                default:
+                    return;
+            }
+            AI.trackEvent('Open in playground initiated');
+            let filename = `script-lab-playground-${environment.current.host}${extension}`;
+            let url = environment.current.config.runnerUrl + `/open-in-playground/${environment.current.host}/${type}/${id}/${filename}`;
+            if (isDownload) {
+                window.open(url, '_blank');
```

Comment:
I agree think that the filesaver provides a better user experience.  Though on the other hand, I also want to have a link that can "just work" as well as a URL (rather than a web api call that returns a json with the data, for example).  So I'm fine keeping it as is, if we can check that it works on Safari (the others I assume it will...)